04week/mastermind.js

```diff
@@ -28,12 +28,29 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(guess) {
   // your code here
+  let sameSpot = 0;
+  let solutionArr = solution.split("");
+  let guessArr = guess.split("");
+  for(let i=0; i < solutionArr.length; i++) {
+  if(solutionArr[i] === guessArr[i]) {
+    sameSpot = sameSpot + 1;
```

Comment:
This is also only have the logic, where you are detecting if the got the letter in the right position, but you still need to handle getting the correct letter but in the wrong position.